webui/react/src/components/CreateExperimentModal.tsx

```diff
@@ -0,0 +1,88 @@
+import { Alert, Modal } from 'antd';
+import yaml from 'js-yaml';
+import React, { SetStateAction, useCallback, useState } from 'react';
+import MonacoEditor from 'react-monaco-editor';
+
+import { routeAll } from 'routes';
+import { forkExperiment } from 'services/api';
+
+import css from './CreateExperimentModal.module.scss';
+
+export interface CreateExpModalState {
+  visible: boolean;
+  config: string;
+}
+
+interface Props {
+  title: string;
+  okText: string;
+  parentId: number;
+  state: CreateExpModalState;
+  setState: (arg0: SetStateAction<CreateExpModalState>) => void;
+}
+
+const CreateExperimentModal: React.FC<Props> = (
+  { state, setState, parentId, ...props }: Props,
+) => {
+  const [ configError, setConfigError ] = useState<string>();
+
+  const editorOnChange = useCallback((newValue: string) => {
+    setState((existingState: CreateExpModalState) => ({ ...existingState, config: newValue }));
+    setConfigError(undefined);
+  }, [ setState, setConfigError ]);
+
+  const monacoOpts = {
+    minimap: { enabled: false },
+    selectOnLineNumbers: true,
+  };
```

Comment:
**nit:** I've just been putting this directly into the `MonacoEditor` component. Then no need to play connect the dots when looking for it.

Reply:
**chore:** add `scrollBeyondLastLine: false,` to prevent it from scrolling to show a bunch of white space below the last line.